frontend/searching.cpp

```diff
@@ -0,0 +1,79 @@
+#include "mainwindow.h"
```

Comment:
Need the copyright header in this and the other files that have been added in this PR.